File: src/Hippo.java
public class Hippo extends Pachyderm {

    Hippo(){
        super(true, "vegetation", "savannah", true, "Harold", 4, 0);
    }

    /**
     *  Constructor with only Animal parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     */
    Hippo(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs){
        super(awake, diet, habitat, hungry, name, numLegs);
    }

    /**
     *  Constructor with Animal and Pachyderm parameters
     * @param awake:boolean - Is the animal awake?
     * @param diet:String - What does the animal eat?
     * @param habitat:String - Where does the animal live?
     * @param hungry:boolean - Is the animal hungry?
     * @param name:String - The name of the animal
     * @param numLegs:int - How many legs does the animal have?
     *
     * @param numTusks:int - How many tusks does the animal possess?
     */
    Hippo(boolean awake, String diet, String habitat, boolean hungry, String name, int numLegs, int numTusks){
        super(awake, diet, habitat, hungry, name, numLegs, numTusks);
    }

    /**
     * Overwrite roam for Hippos
     */
    @Override
    public void roam() { System.out.println(name + " the Hippo: Sinks into his pond."); }

    /**
     * Overwrite make noise for Hippos (5 different noisses)
     * 20% of the time grunts, groans, growls, wheezes, or
     */
    @Override
    public void makeNoise() {
        double rand = Math.random();
        if (rand < 0.2) {
            System.out.println(name + " the Hippo: *grunts*");
        } else if (rand < 0.4){
            System.out.println(name + " the Hippo: *groans*");
        } else if (rand < 0.6) {
            System.out.println(name + " the Hippo: *growls*");
        } else if (rand < 0.8) {
            System.out.println(name + " the Hippo: *wheezes*");
        } else {
            System.out.println(name + " the Hippo: *chuffs*");
        }
    }
}

File: Project2/Hippo.py
import random
from Pachyderm import Pachyderm

class Hippo(Pachyderm):

  '''
    Generic initializer for Hippo
    @param name:String, default:'Harold'
    @param awake:Boolean, default:False
    @param hungry:Boolean, deault:True
    @param num_legs:Int, default:4
  '''
  def __init__(self, name='Harold', awake=False, hungry=True, num_legs=4):
    super(Hippo, self).__init__(name, awake, hungry, num_legs)

  '''
    Hippo has an equal chance to make one of 5 sounds
  '''
  def make_noise(self):
    animal_type = type(self).__name__
    random_number = random.random()
    if random_number < 0.2:
      print('{} the {} grunts.'.format(self.name, animal_type))
    elif random_number <  0.4:
      print('{} the {} groans.'.format(self.name, animal_type))
    elif random_number < 0.6:
      print('{} the {} growls.'.format(self.name, animal_type))
    elif random_number < 0.8:
      print('{} the {} wheezes.'.format(self.name, animal_type))
    else:
      print('{} the {} chuffs.'.format(self.name, animal_type))
      

